src/app/archive/archive_lib/processor.ml

```diff
@@ -1230,8 +1241,10 @@ module Block = struct
       deferred_result_list_fold user_cmd_ids_and_seq_nos ~init:()
         ~f:(fun () (user_command_id, sequence_no) ->
           Block_and_signed_command.add_if_doesn't_exist
-            (module Conn)
-            ~block_id ~user_command_id ~sequence_no )
+            (module Conn )
+            (* TODO(omerzach): Do we need to thread real values down here? *)
```

Comment:
Need to figure this out

Reply:
I think we have a dependency cycle here :grin: IIRC @psteckler has been holding off implementing the balances for this part until you were done with this PR, so I think you can leave them as TODOs, or even re-attribute the TODOs to him.